contrib: Add installation script for tools in devcontainer (#31534)
cilium-team/cilium: Documentation/contributing/development/dev_setup.rst

Depending on your specific development environment and requirements, you
can follow the detailed instructions below.

> Using ``devcontainer``

Reviewer: Can you move the below to the existing devcontainer section? 

https://docs.cilium.io/en/latest/contributing/development/dev_setup/#dev-container

https://github.com/cilium/cilium/blob/0ac34f084f726e638e75a9a639c5fd1acc6f9897/Documentation/contributing/development/dev_setup.rst?plain=1#L603

Author: Ah, sorry i missed that. that is much better. fixed in https://github.com/cilium/cilium/pull/31534/commits/e4b8b7a2708a039446af44d41061abc997a51e22

The container is based on the official Cilium builder image and provides all the dependencies
required to build Cilium.

> You can also install the required packages with ``contrib/scripts/devcontainer-setup.sh``.

Reviewer: the required packages -> the common packages (e.g. kind, kubectl, and cilium-cli)

Author: https://github.com/cilium/cilium/pull/31534/commits/b6b694693a6d41ba6b2b5f2650407f2244ae4a01 fixed.

> You can also install the common packages (e.g. kind, kubectl, and cilium-cli) with ``contrib/scripts/devcontainer-setup.sh``.

Reviewer 2: ```suggestion
You can also install common packages, such as kind, kubectl, and cilium-cli, with ``contrib/scripts/devcontainer-setup.sh``:
```

.. code-block:: shell-session

    $ ./contrib/scripts/devcontainer-setup.sh

> Depending on your requirement, you can modify the package versions as you like.
> This needs to be set up only once when you create the ``devcontainer`` for the cilium development.

Reviewer 2: ```suggestion
Package versions can be modified to fit your requirements.
This needs to only be set up once when the ``devcontainer`` is first created.
```
